### DISCUSSION TITLE
Project announcement: DSJAS-CLI

### DISCUSSION THREAD

**Discussion by ejv2** [DSJAS/DSJAS#22] [General]
As a quick side project, I am working on a DSJAS command-line interface, which will allow developers and power users to manage a DSJAS install using a handy tool in the terminal. In its current state, it only runs on UNIX-like platforms (Linux, *BSDs, Mac-OS), but there isn't really a reason why it *can't* run on Windows. It would just be significantly more difficult for me to test/develop for such platforms, as I don't really work outside of Linux/OpenBSD any more.

On release, the GitHub repository will be located at https://github.com/DSJAS/DSJAS-CLI/. I've written the whole thing from scratch in C99 (with the help of a couple of file parsing libraries to cut out some repetetive INI/JSON parsing code).

Here's a quick screenshot of one of the currently implemented commands:
![DSJAS Terminal Screenshot](https://i.imgur.com/XmajGYU.png "Screenshot of the config summary command")

So far, planned features are the following:

1. Configuration file parsing, querying and dynamic modification (enable/disable themes from the command line)
2. Developer tools to create, modify and manage themes, modules and extensions from the command line
3. Possible automatic upgrades of DSJAS installs

Effectively, I'm just going to be writing a program that automates some of the things I need to do when I am writing the program. Any thoughts, feedback or suggestions of things to include would be brilliant. As it may have been evident by the "planned features" list, we are a little light on things to include right now. Anything that you may find useful will be welcome to go into the program.

Thoughts?

**Comment by SinisterSpatula** [comment]
If the transactions system gets more built up for realism with a set of transaction names and amounts that can be inserted into the Leger, like point of sale purchase, car payment, rent, utilities, subscriptions etc...  could be useful to have a CLI command that can refresh those for a given account or all accounts.  (Along with corresponding button on the dashboards to do it also).  So users no longer have to populate transactions manually.  Should be fixed type and variable.  (So the system can know if this kind of transaction should always be the same amount and only appear once per month, or if its a common random type that should be within a stored range of values.)  And which type of account the transaction may belong to (checking, savings, loan, etc)

**Comment by ejv2** [comment]
Getting close to a version v1.0. Once the creation tool is completed, I am happy to release this publicly. Should be at least a few more weeks, though.

**Reply by ejv2** [reply]
I've gone ahead and made this public and will be doing the rest of development publicly. In-progress versions should now be available.

**Comment by ejv2** [comment]
Closed as stale
